### ISSUE TITLE
Is it possible to finetune BPE embeddings along with the rest of my network?

### ISSUE THREAD

**Issue by JoaoLages** [soaxelbrooke/python-bpe#8]
I'd like to train it along with the other weights of my model, in Pytorch if possible

**Comment by soaxelbrooke** [comment]
Hey @JoaoLages! Are you talking about using the gradient information from the backprop to learn better splits? Or about learning the embeddings that the byte pair tokens are mapped to?

**Comment by JoaoLages** [comment]
Second option! I want to use pre-trained BPEembeddings and update the byte pair tokens that were used in my batch.

**Comment by soaxelbrooke** [comment]
This should be handled in your NN library, as it likely has no idea what actually strings your tokens map to (just the embedding indexes that are input to the model).  You should just be able to initialize your embeddings with the pretrained ones you have, leave them unfrozen (the default), and train you NN.  So, no adjustments needed here, since this library only handles learning the byte pair encoding vocab and transforming text into the embedding indexes.

**Comment by JoaoLages** [comment]
Alright, though it had options to load default pretrained embeddings. Thank you for your quick response!
